virt/xen/xen.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python
+
+import sys
+from time import time
+import XenAPI
+from XenAPI import Failure, NewMaster
+from collections import defaultdict, OrderedDict
+import virt
+
+class Guest(object):
```

Comment:
I really really really tried to do a pull request against tag:14, but ended up doing it against master. virt.Guest is not present in version 14, which is what I'm using (is what RH has as stable in RH7). There are things I half-backported, most I thought would be useful, and some I wasn't entirely sure what they do. I do watn, however, to clean the code and make it as good as possible for this project.